### ISSUE TITLE
Polls are not reflected correctly in Threads

### ISSUE THREAD

**Issue by kittykat** [vector-im/element-web#20282]
### Steps to reproduce

1. Create a poll, then open it in threads view (e.g. start replying in a thread)

1. Someone else creates a poll and ends it
2. Start a thread with the poll as a root message
3. See that I can still "vote" in it

#### What did you expect?
Threads reflects the correct state of polls

#### What happened instead?

![Screenshot from 2021-12-20 17-26-21](https://user-images.githubusercontent.com/51663/146809349-6766aac5-6108-486e-a5e8-ec478ae637d7.png)

![Screenshot from 2021-12-20 17-24-04](https://user-images.githubusercontent.com/51663/146809360-b7388071-9d48-481c-84ea-e273221b3704.png)

### Browser information

Chromium 96.0.4664.45 (Official Build) Arch Linux (64-bit)

### URL for webapp

develop.element.io

### Application version

Element version: dafdc1d64215-react-2330292c24f5-js-2aae2362e3a8 Olm version: 3.2.8

### Will you send logs?

No

**Comment by daniellekirkwood** [comment]
Verified with Germain today that this is still an issue: Users can still update their answers to a thread once it's ended...

**Comment by daniellekirkwood** [comment]
Hey @andybalaam - could you help us with this? 
On investigation it looks like Polls "End" don't take effect when/if there's a Thread attached to it...

**Comment by andybalaam** [comment]
Yes, I will add it to the work we have for polls, thank you @daniellekirkwood

**Comment by andybalaam** [comment]
Hmm, vote events don't seem to show up in threads when I show hidden events:

![image](https://user-images.githubusercontent.com/76812/152150257-f908b0d6-4196-45a4-8add-69b8512c7510.png)

**Comment by gsouquet** [comment]
Two other issues likely have the same root cause as what is causing the problem with polls

- https://github.com/vector-im/element-web/issues/20848
- https://github.com/vector-im/element-web/issues/20849
